### PR TITLE
Using latest checklist items and groups api changes.

### PR DESCRIPTION
**Pull request by ngerakines**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -11,10 +11,15 @@ var request = require('request'),
 // defaults
 this._config = { host: baseURI };
 
+function parseChecklistPayload(input) {
+  return parsePayload("title", input)
+}
+
 // prepare payloads
-function parsePayload(input) {
+function parsePayload(key, input) {
```
Comment (ngerakines) [inline index.js:19]: This is a small experiment to create shortcut functions for specific payloads. Now that item and group creation only requires either a label or display text, the following methos should work:

``` javascript
function parseItemPayload(input) { return parsePayload("display_text", input); }
```

File: test/mocks.js [hunk 2]
```diff
@@ -1,24 +1,44 @@
 this.raw = function(method, endpoint, payload, callback) {
-    if (arguments.length === 3) {
-      callback = payload;
-    }
-    callback(null, 200, {method: method, endpoint: endpoint, payload: payload });
-  };
+  if (arguments.length === 3) {
+    callback = payload;
+  }
+  switch(method) {
```
Comment (ngerakines) [inline test/mocks.js:5]: Added this switch to test default "success" return codes for GET, PUT, POST, PATCH and DELETE operations. Now that all creation requests are treated equally, there are fewer edge cases to deal with.

**Comment by ngerakines** [comment]
I thought about adding a url builder object, so you could do `ninja.urls.Checklists()` or `ninja.urls.Checklist(checklistId)`, but wanted to ask about the best way to do it.
